tests/testthat/test-lambda-path.R

```diff
@@ -1,44 +1,153 @@
-context("lambda path computations")
+context("lambda path")
 
-test_that("lambda paths are computed appropriately", {
+test_that("lambda paths are computed as in glmnet", {
   library(glmnet)
   glmnet.control(fdev = 0)
 
-  n <- 100
-  d <- 3
-
-  # TODO(jolars): test for sparse with standardization once in place centering
-  # has been implemented
-
-  for (sparse in c(TRUE, FALSE)) {
-    x <- Matrix::rsparsematrix(n, d, density = 0.2)
-    if (!sparse)
-      x <- as.matrix(x)
-
-    for (alpha in c(0, 0.5, 1)) {
-      # TODO(jolars): figure out why glmnet is returning such odd
-      # lambda paths when no intercept is fit
-      for (intercept in c(TRUE)) {
-        for (family in c("gaussian")) {
-          set.seed(1)
-          y <- switch(family,
-                      gaussian = rnorm(n, 5, 2),
-                      binomial = rbinom(n, 1, 0.2))
-
-          sfit <- sgdnet(x, y,
-                         family = family,
-                         alpha = alpha,
-                         standardize = !sparse,
-                         intercept = intercept)
-          gfit <- glmnet(x, y,
-                         family = family,
-                         alpha = alpha,
-                         standardize = !sparse,
-                         intercept = intercept)
-
-          expect_equal(sfit$lambda, gfit$lambda)
-        }
+  # TODO(jolars): test for sparse features with standardization once in-place
+  # centering has been implemented
+
+  n <- 1000
+  p <- 2
+  set.seed(1)
+
+  grid <- expand.grid(
+    family = c("gaussian", "binomial", "multinomial"),
+    intercept = TRUE,
+    sparse = c(TRUE, FALSE),
+    alpha = c(0, 0.5, 1),
+    standardize = c(TRUE, FALSE),
+    stringsAsFactors = FALSE
+  )
+
+  x <- Matrix::rsparsematrix(n, p, 0.5)
+
+  for (i in seq_len(nrow(grid))) {
+    pars <- list(
+      x = if (grid$sparse[i]) x else as.matrix(x),
+      standardize = if (grid$sparse[i]) FALSE else grid$standardize[i],
+      family = grid$family[i],
+      intercept = grid$intercept[i],
+      alpha = grid$alpha[i],
+      thresh = 1e-1 # we only care about the regularization paths
+    )
+
+    pars$y <- switch(pars$family,
+                     gaussian = rnorm(n, 10, 2),
+                     binomial = rbinom(n, 1, 0.8),
+                     multinomial = rbinom(n, 3, 0.5))
+
+    sfit <- do.call(sgdnet, pars)
+    gfit <- do.call(glmnet, pars)
+
+    expect_equal(sfit$lambda, gfit$lambda)
```

Comment:
Is there a reason you aren't testing `coef(sfit)` and `coef(gfit)` equal here (or elsewhere)? I think it's probably useful to check that pathwise fitting gets the right result (in addition to sequential fitting) to protect against getting something wrong vs warm-starting.

Reply:
Since coef(fit) is the same as predict(fit, type = "coefficients") it is actually being tested for instance here:

https://github.com/jolars/sgdnet/blob/134d09d54b63bb2e925b2c919f713c5e375afa4b/tests/testthat/test-families.R#L40-L44

The purpose of `test-lambda-path.R` is only to check the calculations of the lambda max and the lambda path, not the actual fits from them, which we do in `test-families.R` and elsewhere.